modules/assim.sequential/R/sda.enkf_MultiSite.R

```diff
@@ -164,41 +153,30 @@ sda.enkf.multisite <- function(settings,
   }
   obs.times <- obs.times.POSIX
   read_restart_times <- c(lubridate::ymd_hms(start.cut, truncated = 3), obs.times)
-  #obs.times <- c(start.cut, obs.times)
-  nt          <- length(obs.times)
+  nt  <- length(obs.times) #sets length of for loop for Forecast/Analysis
   if (nt==0) PEcAn.logger::logger.severe('There has to be at least one Obs.')
-  bqq         <- numeric(nt + 1)
-  
 
-  ###-------------------------------------------------------------------###
-  ### If this is a restart - Picking up were we left last time          ###
-  ###-------------------------------------------------------------------###----   
-  if (restart){
-    
-    if(!file.exists(file.path(settings$outdir,"SDA", "sda.output.Rdata"))) PEcAn.logger::logger.severe("The SDA output from the older simulation doesn't exist.")
-    load(file.path(settings$outdir,"SDA", "sda.output.Rdata"))
-    
-    #this is where the old simulation will be moved to
-    old.dir <- lubridate::year(names(FORECAST)) %>% tail(1)
-    #--- Updating the nt and etc
-    if(!dir.exists(file.path(settings$outdir,"SDA",old.dir))) dir.create(file.path(settings$outdir,"SDA",old.dir))
-    # finding/moving files to it's end year dir
-    files.last.sda <- list.files(file.path(settings$outdir,"SDA"))
-    #copying
-    file.copy(file.path(file.path(settings$outdir,"SDA"),files.last.sda),
-              file.path(file.path(settings$outdir,"SDA"),paste0(old.dir,"/",files.last.sda))
-    )
-    params<-new.params
-    sim.time <-2:nt # if It's restart I added +1 from the start to nt (which is the last year of old sim) to make the first sim in restart time t=2
-    
-    X <-FORECAST[[length(FORECAST)]]
+# Model Specific Setup ----------------------------------------------------
+
+  #--get model specific functions
+  do.call("library", list(paste0("PEcAn.", model)))
```

Comment:
1. R will (correctly) complain that `library` calls shouldn't exist in package code. Could this one be eliminated by having the package name be part of the `my.<fn>` strings below? e.g. `my.write_restart <- paste0("PEcAn.", model, "::write_restart.", model)
2. This section will break if any model package follows PEcAnAssimSequential's lead and removes the `.` from its name. Rather than hard-code the assumption that every package name takes the form `PEcAn.<modelname>` into so many different lines of the file, would  it be possible to move the construction of the package name someplace further upstream and pass that *package* name around instead of repeatedly pasting it together?


(R's package checks also will complain about `library` calls in package code.

Reply:
Will integrate your suggested fix and do some test runs to ensure the function calls will work.